OpenRA.Mods.Common/SpriteLoaders/PngSheetLoader.cs

```diff
@@ -92,22 +92,32 @@ public bool TryParseSprite(Stream s, out ISpriteFrame[] frames, out TypeDictiona
 			return true;
 		}
 
-		void RegionsFromFrames(Png png, out List<Rectangle> regions, out List<float2> offsets)
+		void RegionsFromFrames(Stream s, Png png, out List<Rectangle> regions, out List<float2> offsets)
 		{
 			regions = new List<Rectangle>();
 			offsets = new List<float2>();
+			var pngRectangle = new Rectangle(0, 0, png.Width, png.Height);
 
 			string frame;
 			for (var i = 0; png.EmbeddedData.TryGetValue("Frame[" + i + "]", out frame); i++)
 			{
 				// Format: x,y,width,height;offsetX,offsetY
 				var coords = frame.Split(';');
-				regions.Add(FieldLoader.GetValue<Rectangle>("Region", coords[0]));
+				var region = FieldLoader.GetValue<Rectangle>("Region", coords[0]);
+				if (!pngRectangle.Contains(region))
+				{
+					var fileStream = s as FileStream;
+					var fileName = fileStream != null ? fileStream.Name : "unknown file";
```

Comment:
Summarising the discussion from IRC: hardcoding specializations for different stream types here doesn't sit well with me. I would prefer it if we could:
 * Throw a fairly simple exception here (e.g. just "Invalid frame regions defined")
 * Add similar checks to the `PngSheetImportMetadataCommand` that report the full details on error
 * Add a `try`/`catch` around the `ParseSequences` call in `CheckSquenceSprites` and report the sequence name/yaml line that references the invalid file.

Reply:
Went with https://github.com/OpenRA/OpenRA/pull/17923.

Reply:
```csharp
try
{
	modData.SpriteSequenceLoader.ParseSequences(modData, ts, sc, n);
}
catch
{
	throw new InvalidDataException(n.Key + " " + n.Location + " " + n.Value);
}
```

will still crash in `OpenRA.Mods.Common.SpriteLoaders.PngSheetLoader.TryParseSprite` with exactly the same not helpful exception.

Reply:
Also went with the simple exception here. An exception with slightly more context is better than nothing although it is worse without the filename.